### PR TITLE
Bug Fixes and Enhancements from LDMX Project Development

### PR DESCRIPTION
**Pull request by bengineerd**
### Description

There are many small changes, mostly to fix issues found in simulation.

### Details

- AxiStreamResize
  - Better error reporting if generic assert fails
- Ad9249ReadoutGroup2
  - Initialize signal to avoid unknowns in simulation
- Ad9249Group (simulation)
  - Fix incomplete sensitivity list
- Pgp2FcAxi
  - Apply TX FC bus from input to PGP block
- PGP2FC GTY IP Core
  - Remove "project_parameters" from XCI file
  - Allows XCI to load with different US+ FPGAs
- PGP2FC GTY IP Core Wrapper
  - Output `rxPmaResetDone`
    - Needed for external MMCM reset
  - Update resets based on simulation discoveries
  - Set freerun frequency to 185.71/2 MHz
- Add GTH IP Core and wrapper for PGP2FC
- ClockManagerUltraScale (Plus)
  - Resolve unknowns on DrpDo output from GT to zeros to avoid propagating them in simulation
- GtRxAlignCheck
  - Use registered values for read-only AXI-Lite registers
  - Apply bug fix so `resetErr` only reset the FSM when `resetDone` is high
- AxiLiteRingBuffer
  - Fix bugs in Rogue Device
- Rogue Devices
  - Add several registers to the "NoConfig" group

### CODE REVIEW

File: xilinx/UltraScale+/clocking/rtl/ClockManagerUltraScale.vhd [hunk 1]
```diff
@@ -156,7 +157,9 @@ begin
          drpWe           => drpWe,
          drpAddr         => drpAddr,
          drpDi           => drpDi,
-         drpDo           => drpDo);
+         drpDo           => drpDo01);
+
+   drpDo01 <= to_stdLogicVector(to_bitvector(drpDo));
```
Comment (ruck314) [inline xilinx/UltraScale+/clocking/rtl/ClockManagerUltraScale.vhd:162]: Why was this change needed?

File: axi/axi-stream/rtl/AxiStreamResize.vhd [hunk 2]
```diff
@@ -88,7 +88,10 @@ begin
    -- Make sure data widths are appropriate.
    assert ((SLV_BYTES_C >= MST_BYTES_C and SLV_BYTES_C mod MST_BYTES_C = 0) or
            (MST_BYTES_C >= SLV_BYTES_C and MST_BYTES_C mod SLV_BYTES_C = 0))
-      report "Data widths must be even number multiples of each other" severity failure;
+      report "Data widths must be even number multiples of each other" & LF &
+      "SLV_BYTES_C= " & integer'image(SLV_BYTES_C) & LF &
+      "MST_BYTES_C= " & integer'image(MST_BYTES_C)
+      severity failure;
```
Comment (ruck314) [inline axi/axi-stream/rtl/AxiStreamResize.vhd:94]: Where does `LF` get defined?

File: xilinx/general/rtl/GtRxAlignCheck.vhd [hunk 3]
```diff
@@ -230,8 +224,6 @@ begin
             end if;
          ----------------------------------------------------------------------
          when READ_S =>
-            -- Reset the flag
-            v.locked := '0';
```
Comment (ruck314) [inline xilinx/general/rtl/GtRxAlignCheck.vhd:234]: Why is locked != 0 for READ_S state anymore?

File: axi/axi-stream/rtl/AxiStreamResize.vhd (hunk 2 above)
Comment (bengineerd) [reply axi/axi-stream/rtl/AxiStreamResize.vhd:94]: It is part of the VHDL standard package (included in everything by default). The same place `integer` and such are defined.

```        
    type CHARACTER is (
             NUL, SOH, STX, ETX, EOT, ENQ, ACK, BEL,
             BS,  HT,  LF,  VT,  FF,  CR,  SO,  SI, 
             DLE, DC1, DC2, DC3, DC4, NAK, SYN, ETB,
             CAN, EM,  SUB, ESC, FSP, GSP, RSP, USP,
         
             ' ', '!', '"', '#', '$', '%', '&', ''',
             '(', ')', '*', '+', ',', '-', '.', '/',
             '0', '1', '2', '3', '4', '5', '6', '7',
             '8', '9', ':', ';', '<', '=', '>', '?',
         
             '@', 'A', 'B', 'C', 'D', 'E', 'F', 'G',
             'H', 'I', 'J', 'K', 'L', 'M', 'N', 'O',
             'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W',
             'X', 'Y', 'Z', '[', '\', ']', '^', '_',
         
             '`', 'a', 'b', 'c', 'd', 'e', 'f', 'g', 
             'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o',
             'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 
             'x', 'y', 'z', '{', '|', '}', '~', DEL);
             ```

File: xilinx/UltraScale+/clocking/rtl/ClockManagerUltraScale.vhd (hunk 1 above)
Comment (bengineerd) [reply xilinx/UltraScale+/clocking/rtl/ClockManagerUltraScale.vhd:162]: Some DRP registers seem to initialize with 'U' or 'X' values. This was a problem simulation as the 'U' and 'X' propagated to an MGT and then caused havoc. Forcing them to resolve to 0 (as they would in real hardware) fixed the issue.

This was discovered when using Rogue software to adjust the phase delay of an MMCM clock via the DRP.

File: axi/axi-stream/rtl/AxiStreamResize.vhd (hunk 2 above)
Comment (ruck314) [reply axi/axi-stream/rtl/AxiStreamResize.vhd:94]: Thanks.  Good to know

File: xilinx/general/rtl/GtRxAlignCheck.vhd (hunk 3 above)
Comment (bengineerd) [reply xilinx/general/rtl/GtRxAlignCheck.vhd:234]: It was done in this commit.
https://github.com/slaclab/surf/commit/7bf95215ebe67b2a3a16bf04396edda3acf00d2d

Christos was applying a bug fix created by Julian and deployed in the lcls-timing-core repo.
The changes we applied to match https://github.com/slaclab/lcls-timing-core/blob/7c77d7502e270c8f352054cfb071a7a675bc1a47/LCLS-II/core/rtl/GthRxAlignCheck.vhd
In this code, `v.locked := '0'` is the default assignment, so having it also assigned in the state machine was redundant.

**Review by ruck314 — APPROVED** [review]

File: python/surf/axi/_AxiLiteRingBuffer.py [hunk 4]
```diff
@@ -28,59 +28,59 @@ def __init__(self, datawidth=32, **kwargs):
         ##############################
 
         self.add(pr.RemoteVariable(
-            name         = 'blen',
+            name         = 'BufferLength',
```
Comment (ruck314) [inline python/surf/axi/_AxiLiteRingBuffer.py:31]: @bengineerd Isn't this change (and the other remote variable name changes) going to break existing software?
